tests/vlan/test_autostate_disabled.py

```diff
@@ -0,0 +1,120 @@
+        if not res.is_successful:
+            self.restore_interface_admin_state(duthost, ifs_status)
+            pytest.fail('shutdown "{vlan_members}" in {vlan} failed'.format(vlan_members=vlan_members, vlan=vlan))
+        logging.info('waiting for "{vlan_members}" shutdown in {vlan}'.format(vlan_members=vlan_members, vlan=vlan))
+        if not wait_until(60, 5, 0, self.check_interface_oper_state, duthost, vlan_members, "down"):
+            self.restore_interface_admin_state(duthost, ifs_status)
+            pytest.fail('shutdown "{vlan_members}" in {vlan} failed'.format(vlan_members=vlan_members, vlan=vlan))
+
+        # Check whether the oper_state of vlan interface is changed as expected.
+        ip_ifs = duthost.show_ip_interface()['ansible_facts']['ip_interfaces']
+        if len(vlan_available) > 1:
+            # If more than one vlan comply with the above test requirements, then there are members in other vlans
+            # that are still up. Therefore, the bridge is still up, and vlan interface should be up.
+            if ip_ifs.get(vlan, {}).get('oper_state') != "up":
+                self.restore_interface_admin_state(duthost, ifs_status)
+                pytest.fail('vlan interface of {vlan} is not up as expected'.format(vlan=vlan))
+        else:
+            # If only one vlan comply with the above test requirements, then all the vlan members across all the vlans
+            # are down. Therefore, the bridge is down, and vlan interface should be down.
+            if ip_ifs.get(vlan, {}).get('oper_state') != "down":
+                self.restore_interface_admin_state(duthost, ifs_status)
+                pytest.fail('vlan interface of {vlan} is not down as expected'.format(vlan=vlan))
+
+        # Restore all interfaces to their original admin_state.
+        self.restore_interface_admin_state(duthost, ifs_status)
+
+    def restore_interface_admin_state(self, duthost, ifs_status):
+        """
+        Restore all interfaces to their original admin_state at the end of test.
+        """
+        ifs_up, ifs_down = [], []
+        for interface in ifs_status:
+            admin_state = ifs_status[interface].get('admin')
+            if admin_state == 'up':
+                ifs_up.append(interface)
+            elif admin_state == 'down':
+                ifs_down.append(interface)
+        if len(ifs_up) > 0:
+            res = duthost.no_shutdown_multiple(ifs_up)
+            if not res.is_successful:
+                logging.error('startup "{interfaces}" on DUT {hostname} failed'.
+                              format(interfaces=ifs_up, hostname=duthost.hostname))
+        if len(ifs_down) > 0:
+            res = duthost.shutdown_multiple(ifs_down)
+            if not res.is_successful:
+                logging.error('shutdown "{interfaces}" on DUT {hostname} failed'.
+                              format(interfaces=ifs_down, hostname=duthost.hostname))
+
+    def get_interface_status(self, duthost):
```

Comment:
How about move this code the sonic host?

Reply:
Yes, I have moved this code and wrote an related doc here: https://github.com/Azure/sonic-mgmt/blob/4d4db8e6844677a46c79070aff5670151b295b61/docs/api_wiki/sonichost_methods/get_interfaces_status.md